feat(query): binary join across downsample and raw cluster

TanviBhavsar opened the pull request:
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 
Currently Binary join is not done across downsample and raw cluster. If query time is greater than raw retention, the entire query will be sent to downsample cluster and raw cluster both and then the results will be stitched.

**New behavior :**
If LHS/RHS query start time is less than raw retention we will send only LHS/RHS to downsample cluster, RHS/LHS would be evaluated on raw cluster and then Binary join will be computed using InprocessDispatcher
Example:  `some_metric offset 8d - some_metric`
 If raw retention is 7 days, LHS will be sent o downsample cluster and RHS will be evaluated on raw cluster and then Binary Join will be applied

vishramachandran commented on coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangePlanner.scala:
```diff
@@ -90,12 +93,21 @@ class LongTimeRangePlanner(rawClusterPlanner: QueryPlanner,
     val onKeysReal = ExtraOnByKeysUtil.getRealOnLabels(logicalPlan, queryConfig.addExtraOnByKeysTimeRanges)
 
+
+    val dispatcher = if (lhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher] &&
+      rhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher]) {
+      val lhsCluster = lhsExec.dispatcher.asInstanceOf[ActorPlanDispatcher].clusterName
+      val rhsCluster = rhsExec.dispatcher.asInstanceOf[ActorPlanDispatcher].clusterName
+      if (rhsCluster.equals(lhsCluster)) PlannerUtil.pickDispatcher(lhsExec.children ++ rhsExec.children)
+      else inProcessPlanDispatcher
+    } else inProcessPlanDispatcher
```
`clusterName`  should be in PlanDispatcher trait. This will avoid the type casts.

TanviBhavsar replied:
Added clusterName to PlanDispatcher

vishramachandran reviewed (approved):
Please remove instanceof checks in

```
val dispatcher = if (lhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher] &&
rhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher]) {
```

TanviBhavsar commented:
> Please remove instanceof checks in
> 
> ```
> val dispatcher = if (lhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher] &&
> rhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher]) {
> ```

Removed and added isLocalCall method as discussed
